src/Proto3/Suite/DotProto/Parsing.hs

```diff
@@ -19,12 +22,25 @@ import Text.Parser.Char
 import Text.Parser.Combinators
 import Text.Parser.LookAhead
 import Text.Parser.Token
+import qualified Turtle
 
 ----------------------------------------
--- interface
-
-parseProto :: String -> Either ParseError DotProto
-parseProto = parse topLevel "" . stripComments
+-- interfaces
+
+-- | @parseProto mp inp@ attempts to parse @inp@ as a 'DotProto'. @mp@ is the
+-- module path to be injected into the AST as part of 'DotProtoMeta' metadata on
+-- a successful parse.
+parseProto :: Path -> String -> Either ParseError DotProto
+parseProto modulePath = parse (topLevel modulePath) "" . stripComments
```

Comment:
Yeah, it really shouldn't be `String`, but there's a comment:

```-- | This module contains a near-direct translation of the proto3 grammar
--   It uses String for easier compatibility with DotProto.Generator, which needs it for not very good reasons
```

It's on the list of things to improve, but (a) we're likely to obsolete this parser and (b) I don't want to couple that change with this PR, even if it's likely to be easy.